### PR TITLE
Add ability to specify Terraform Cloud Project in cloud block

### PR DESCRIPTION
**Pull request by joekarl**
Adds project configuration to the workspaces section of the cloud block. Also configurable via the `TF_CLOUD_PROJECT` environment variable. When a project is configured, the following behaviors will occur:
- `terraform init` with workspaces.name configured will create the workspace in the given project
- `terraform workspace new <name>` with workspaces.tags configured will create workspaces in the given project
- `terraform workspace list` will list workspaces only from the given project

The following behaviors are NOT affected by project configuration
- `terraform workspace delete <name>` does not validate the workspace's inclusion in the given project
- When initializing a workspace that already exists in Terraform Cloud, the workspace's parent project is NOT validated against the given project

Adds tests for cloud block configuration of project 
Update changelog
Update docs

## Target Release

1.6.x

### ENHANCEMENTS
* cloud: Terraform now supports setting the Terraform Cloud project via the cloud block and `TF_CLOUD_PROJECT` environment variable. When a cloud configuration creates a new workspace it will be created in the given project. If the project does not exist it will be created. `terraform workspace list` will also respect listing matching workspaces in the configured project.

### CODE REVIEW

File: internal/cloud/backend.go [hunk 1]
```diff
@@ -178,18 +183,26 @@ func (b *Cloud) PrepareConfig(obj cty.Value) (cty.Value, tfdiags.Diagnostics) {
 	}
 
 	WorkspaceMapping := WorkspaceMapping{}
+
+	// Initially set the project via env var
+	WorkspaceMapping.Project = os.Getenv("TF_CLOUD_PROJECT")
```
Comment (uturunku1) [inline internal/cloud/backend.go:188]: I don't think we need line 188 and line 197 to line 199.  They are adding attributes to a temporary WorkspaceMapping object, which get created to call the `Strategy` method (line 195). But the Strategy method is not validating against a wm.Project

File: internal/cloud/backend.go [hunk 2]
```diff
@@ -178,18 +183,26 @@ func (b *Cloud) PrepareConfig(obj cty.Value) (cty.Value, tfdiags.Diagnostics) {
...
 		if val := workspaces.GetAttr("name"); !val.IsNull() {
 			WorkspaceMapping.Name = val.AsString()
 		}
+		if val := workspaces.GetAttr("project"); !val.IsNull() {
+			WorkspaceMapping.Project = val.AsString()
+		}
 		if val := workspaces.GetAttr("tags"); !val.IsNull() {
 			err := gocty.FromCtyValue(val, &WorkspaceMapping.Tags)
 			if err != nil {
 				log.Panicf("An unxpected error occurred: %s", err)
```
Comment (uturunku1) [inline internal/cloud/backend.go:203]: Oops, I see a typo! I know it's not your commit, but could you change that "unxpected" for "unexpected"? 🙏

File: internal/cloud/backend.go [hunk 3]
```diff
@@ -600,16 +638,48 @@ func (b *Cloud) StateMgr(name string) (statemgr.Full, error) {
 	}
 
 	if err == tfe.ErrResourceNotFound {
-		// Create a workspace
-		options := tfe.WorkspaceCreateOptions{
+
+		// Worksapce Create Options
```
Comment (uturunku1) [inline internal/cloud/backend.go:642]:
```suggestion
		// Workspace Create Options
```

File: internal/cloud/backend.go [hunk 4]
```diff
@@ -600,16 +638,48 @@ func (b *Cloud) StateMgr(name string) (statemgr.Full, error) {
 	}
 
 	if err == tfe.ErrResourceNotFound {
```
Comment (uturunku1) [inline internal/cloud/backend.go:640]: It looks like in the case when a resource it's found, meaning the workspace already exists, `terraform init` will output `Terraform Cloud has been successfully initialized!`. Which makes sense because the user should reinitialize as many times as they want (a pre-existing workspace). However, if the user were to run `terraform init` on a cloud block configuration that has a workspace name that exists and a project name that does not exist, then the output will also be `Terraform Cloud has been successfully initialized!`. I wonder if rather than silently fail at creating the new project, we should throw a warning. Thoughts?

**Review by uturunku1 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,7 @@
 ENHANCEMENTS:
 * config: Terraform can now track some additional detail about values that won't be known until the apply step, such as the range of possible lengths for a collection or whether an unknown value can possibly be null. When this information is available, Terraform can potentially generate known results for some operations on unknown values. This doesn't mean that Terraform can immediately track that detail in all cases, but the type system now contains the facility for that and so over time we will improve the level of detail generated by built-in functions, language operators, Terraform providers, etc. [GH-33234]
 * jsonplan: Added `errored` field to JSON plan output, indicating whether a plan errored. [GH-33372]
+* cloud: Terraform now supports setting the Terraform Cloud project via the cloud block and `TF_CLOUD_PROJECT` environment variable. When a cloud configuration creates a new workspace it will be created in the given project. If the project does not exist it will be created. `terraform workspace list` will also respect listing matching workspaces in the configured project.
```
Comment (radditude) [inline CHANGELOG.md:6]: Drive-by note: we usually make changelog updates as a separate step after a PR is merged in order to avoid being trapped in merge conflict purgatory!

Comment (joekarl) [reply CHANGELOG.md]: Removed this

**Review by uturunku1 — APPROVED** [review]
The changes look good!

**Comment by bondsb** [comment]
It looks like this CHANGELOG entry didn't make it in.  Can it be added into the release notes?
